src/protocol/kademlia/impl/peer_routing_table_impl.cpp

```diff
@@ -126,29 +136,31 @@ namespace libp2p::protocol::kademlia {
         return true;
       }
       auto replaceablePeerIt =
-          std::find_if(resizedBucket.begin(), resizedBucket.end(),
+          std::find_if(resizedBucket.rbegin(), resizedBucket.rend(),
                        [](const auto &bpi) { return bpi.is_replaceable; });
 
-      if (replaceablePeerIt == resizedBucket.end()) {
+      if (replaceablePeerIt == resizedBucket.rend()) {
         return Error::PEER_REJECTED_NO_CAPACITY;
       }
       auto removedPeer = (*replaceablePeerIt).peer_id;
       bus_->getChannel<events::PeerRemovedChannel>().publish(removedPeer);
-      resizedBucket.erase(replaceablePeerIt);
+      std::advance(replaceablePeerIt, 1);
```

Comment:
I guess, `++replaceablePeerIt` is more human readable. `std::advance` is more situable for change for greater than one step.